enhancements/ingress/lb-subnet-selection-aws.md

```diff
@@ -0,0 +1,597 @@
+[Unmanaged Subnet Annotation Migration Workflow](#unmanaged-subnet-annotation-migration-workflow).
+
+##### Reason #3: The CCM Doesn't Reconcile NLB Subnets Updates after creation.
+
+The CCM doesn't reconcile `service.beta.kubernetes.io/aws-load-balancer-subnets`
+annotation updates to NLBs once it has been created. However, requiring a service
+deletion and recreation will effectuate these changes.
+
+See [CCM Doesn't Reconcile NLB Subnets Updates](#ccm-doesnt-reconcile-nlb-subnets-updates)
+for more details on why the CCM doesn't reconcile these updates.
+
+### Install Config Proposal (Day 1)
+
+TODO: Install time configuration proposal
+
+### Workflow Description
+
+#### Setting IngressController Subnets during Installation Workflow
+
+Setting the default subnets for all IngressController during installation (Day 1):
+
+TODO: Workflow for install time configuration
+
+#### Creating an IngressController with Subnets after Installation Workflow
+
+Creating an IngressController while specifying the subnets for the LoadBalancer-type
+service after installation:
+
+1. A cluster admin creates an IngressController with
+  `spec.endpointPublishingStrategy.loadBalancer.providerParameters.aws.subnets` specified.
+2. The ingress operator creates the LoadBalancer-type Service and sets the
+   `service.beta.kubernetes.io/aws-load-balancer-subnets` annotation with the subnets
+   provided by the user.
+
+#### Updating an existing IngressController with new Subnets Workflow
+
+Updating an existing IngressController with new subnets requires the service
+to be deleted and recreated:
+
+1. A cluster admin edits an IngressController's
+   `spec.endpointPublishingStrategy.loadBalancer.providerParameters.aws.subnets` field.
+2. The Ingress Operator sets the `LoadBalancerProgressing` condition to `Status: True`
+   on the IngressController and provides a message indicating to the cluster admin
+   that they must delete the service.
+3. The cluster admin deletes the LoadBalancer-type service indicated in the
+   `LoadBalancerProgressing` condition message.
+4. The Ingress Operator recreates the LoadBalancer-type service and configures the
+   `service.beta.kubernetes.io/aws-load-balancer-subnets` on the LoadBalancer-type
+   service, thus creating the load balancer with the desired subnets specified by the
+   cluster admin.
+
+#### Unmanaged Subnet Annotation Migration Workflow
+
+Migrating an unmanaged `service.beta.kubernetes.io/aws-load-balancer-subnets`
+service annotation to `spec.endpointPublishingStrategy.loadBalancer.providerParameters.aws.subnets`
+after upgrading to 4.y doesn't require a cluster admin to delete the LoadBalancer-type
+service:
+
+1. Cluster admin confirms that initially `service.beta.kubernetes.io/aws-load-balancer-subnets`
+   is set on the LoadBalancer-type service managed by the IngressController.
+2. The cluster admin upgrades OpenShift to v4.y and the service annotation is not
+   changed or removed.
```

Comment:
As far as I know, they won't be able to upgrade while the IngressController is the Progressing.

Reply:
I may have misunderstood what is happening in step 1 - is it a thing to do or a thing to check?